docs/cookbook.adoc

```diff
@@ -0,0 +1,152 @@
+
+The following stable URL downloads the latest long-term supported (LTS) Linux x64 JDK version 17 JDK.
+
+[source, html]
+----
+https://api.adoptium.net/v3/binary/latest/17/ga/linux/x64/jdk/hotspot/normal/eclipse
+----
+
+Here's an example of using the API to download the latest Java 17 JRE installer for Windows on 64-bit Intel CPUs.
+
+[source,html]
+https://api.adoptium.net/v3/installer/latest/17/ga/windows/x64/jre/hotspot/normal/eclipse?project=jdk
+
+****
+Looking at these links you can see they follows a general
+https://en.wikipedia.org/wiki/Clean_URL[clean URL]
+structure with replaceable variables in the path:
+`https://api.adoptium.net/v3/binary/latest/{feature_version}/ga/{os}/{arch}/{image_type}/hotspot/normal/eclipse`
+
+so downloading for other Java API versions and platforms is simply a case of replacing the variables with the required values
+
+[unordered.stack]
+feature_version:: the Java API version, choose one of "8", "11", "17", "19" and so on.
+os:: the operating system, one of "linux", "windows", "mac", "solaris", "aix", or "alpine-linux".
+arch:: the CPU architecture, one of "x64", "x86", "x32", "ppc64", "ppc64le", "s390x", "aarch64", "arm", "sparcv9", or "riscv64".
+image_type:: product type, choose one of "jdk" or "jre".
+ 
+[NOTE]
+Not all combinations of values are valid, so if there is no Temurin build available for the combination you specify the API will return a `HTTP/1.1 404 Not Found` response.
+****
+
+
+== Example Three: Scripting a download using the Adoptium API
+
+Rather than use a URL in the browser you can download Temurin binaries and check they were received successfully using a shell script, as follows:
+
+[source, bash]
+----
+#!/bin/bash
+set -eu
+
+# Specify the Java version and platform
+API_URL="https://api.adoptium.net/v3/binary/latest/17/ga/linux/x64/jdk/hotspot/normal/eclipse"
+
+# Fetch the archive
+FETCH_URL=$(curl -s -w %{redirect_url} "${API_URL}")
+FILENAME=$(curl -OLs -w %{filename_effective} "${FETCH_URL}")
+
+# Validate the checksum
+curl -Ls "${FETCH_URL}.sha256.txt" | sha256sum -c --status
+
+echo "Downloaded successfully as ${FILENAME}"
+----
+
+Again, the `API_URL` can be adapted for different Java versions etc. as shown in <<example-two,Example Two>>.
+
+
+== Example Three: More complex queries
+
+The API can be supplemented to filter the details response by Java version, platform, and other conditions and can be used to answer detailed queries.
+
+For example, to find the date of the most recent build of Java 19 on Windows 32-bit. The following API call will find all builds of Java 19 or later for Windows 32-bit architecture, sorted with most recent first.
+
+[source,html]
+https://api.adoptium.net/v3/assets/version/%5B19%2C%5D?os=windows&architecture=x32&sort_method=DATE&sort_order=DESC
```

Comment:
One subtility here that may not be obvious to users, sorting by DATE will give you the most recently released and not most recent version, i.e if your version range was 17+, then a 17 released recently would be higher up than a 19 version released a while ago.

The wording may be ambiguous here as people may think they are getting the date of the most recent version

Reply:
ACK. Clarified that is the release date.